src/tree-internal.component.ts

```diff
@@ -62,7 +62,7 @@ export class TreeInternalComponent implements OnInit {
   public isLeftMenuVisible: boolean = false;
 
   public constructor(@Inject(NodeMenuService) private nodeMenuService: NodeMenuService,
-                     @Inject(TreeService) private treeService: TreeService,
+                     @Inject(TreeService) public treeService: TreeService,
```

Comment:
by extracting `switchFoldingType` (see comment above) you don't have to make this field public